src/GraphQL/Resolvers/NameFieldResolver.cs

```diff
@@ -58,40 +59,31 @@ public class NameFieldResolver : IFieldResolver
         /// </summary>
         /// <param name="target"> The type from which you want to get the value. </param>
         /// <param name="name"> Property/method name. </param>
-        /// <returns> Compiled delegate to get the value. </returns>
-        private static Func<IResolveFieldContext, ValueTask<object?>> CreateDelegate(Type target, string name)
+        /// <returns> Compiled field resolver. </returns>
+        private static IFieldResolver CreateResolver(Type target, string name)
         {
             var param = Expression.Parameter(typeof(IResolveFieldContext), "context");
             var source = Expression.MakeMemberAccess(param, _sourcePropertyInfo);
             var cast = Expression.Convert(source, target);
-            var body = CreateAccessorExpression(cast, name, target);
-            // supports return types of T, Task<T> and ValueTask<T>
-            return MemberResolver.BuildFieldResolverInternal(param, body);
+            var instanceLambda = Expression.Lambda(cast, param);
 
-            static Expression CreateAccessorExpression(Expression member, string name, Type target)
+            MemberInfo? member = target.GetProperty(name, BindingFlags.IgnoreCase | BindingFlags.Instance | BindingFlags.Public | BindingFlags.FlattenHierarchy);
```

Comment:
I think it may throw too in case of hiding property (with same name).

Reply:
Added test; no errors for this situation.